runtime/bcutil/dynload.c

```diff
@@ -167,8 +170,23 @@ findLocallyDefinedClass(J9VMThread * vmThread, J9Module * j9module, U_8 * classN
 		}
 	}
 
+	if ((classLoader == javaVM->systemClassLoader)
+		&& (NULL != classLoader->cpEntriesMutex)
+	) {
+		omrthread_monitor_enter(classLoader->cpEntriesMutex);
+		releaseBootLoaderCpMutex = TRUE;
+	}
+
+	if (J9_ARE_NO_BITS_SET(options, J9_FINDCLASS_FLAG_USE_BOOT_LOADER_CP_ENTRIES) || classLoader == javaVM->systemClassLoader) {
```

Comment:
I don't understand this flag, why it's set where it is, and why this is checking for the bit not set.

Reply:
From the caller in the original code: https://github.com/eclipse-openj9/openj9/blob/a9df63885b6706c1afb9b8fdddffab2d06a7fd6f/runtime/vm/classsupport.c#L571

The `classPathEntries` is NULL if it is a non-bootstrap class loader. We are not using `classLoader->classPathEntries`. I believe we can simply skip calling findLocallyDefinedClassFunction if it is a non-bootstrap class loader.

Reply:
Could be. Long ago when we supported J2ME there could be non-bootstrap classloaders with entries, but not anymore.

Reply:
Seems J9_ARE_NO_BITS_SET is the opposite thing to check.